enhancements/baremetal/enable-baremetal-on-other-platforms.md

```diff
@@ -0,0 +1,198 @@
+---
+title: enable-baremetal-on-other-platforms
+authors:
+  - "@asalkeld"
+  - "@sadasu"
+reviewers:
+  - "@hardys"
+  - "@romfreiman"
+  - "@dhellman"
+approvers:
+  - "@hardys"
+creation-date: 2021-08-20
+last-updated: 2021-08-20
+status: implementable
+see-also:
+  - "/enhancements/baremetal/baremetal-provisioning-config.md"
+replaces:
+superseded-by:
+---
+
+# Enable baremetal on other Platforms to support centralized host management
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Baremetal Host API is only available when deploying an OpenShift cluster with the baremetal
+platform (via the IPI or AI (Assisted Installer) workflow). Having the ability to
+manage baremetal hosts from clusters without requiring the cluster to be on baremetal
+would be beneficial to customers.
+
+## Motivation
+
+An initial driver of this feature are the centralized host management use cases
+in edge topologies, which without this feature, is restricted to having the
+central OpenShift cluster deployed on baremetal.
+
+See:
+- https://github.com/openshift/enhancements/blob/master/enhancements/installer/agent-based-installation-in-hive.md
+- https://github.com/openshift/assisted-service/tree/master/docs/hive-integration
+
+### Goals
+
+The specific goals of this proposal are to:
+
+Support the centralized host management use case by partially enabling Baremetal Host API
+on the following platforms:
+- None
+- OpenStack
+- vSphere
```

Comment:
Could you elaborate on why cloud platforms like AWS, GCP, Azure are left out of this enhancement? 

The motivation is not obvious to me, it may not be obvious to others. 

Maybe it's just to limit the scope for now? 

Noticed this since I had someone asking if it would be possible to enable this use-case on AWS

Reply:
@flaper87 the intention was to limit the scope (and test matrix) to on-premise platforms for now - AFAIK the requirements driving this are only for on-premise atm, but it could be expanded in future should the need arise.

Also if we decide to expand the scope of this to enable PXE deployments in future (e.g Provisioning network not `Disabled` ) I could see that being potentially more difficult with provisioning services running on a cloud, vs in the same environment as the target hosts.